nsxt/resource_nsxt_policy_ipsec_vpn_dpd_profile.go

```diff
@@ -0,0 +1,198 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: MPL-2.0 */
+
+package nsxt
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/vsphere-automation-sdk-go/runtime/protocol/client"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/infra"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+)
+
+var iPSecVpnDpdProfileDpdProbeModeValues = []string{
+	model.IPSecVpnDpdProfile_DPD_PROBE_MODE_ON_DEMAND,
+	model.IPSecVpnDpdProfile_DPD_PROBE_MODE_PERIODIC,
+}
+
+func resourceNsxtPolicyIPSecVpnDpdProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceNsxtPolicyIPSecVpnDpdProfileCreate,
+		Read:   resourceNsxtPolicyIPSecVpnDpdProfileRead,
+		Update: resourceNsxtPolicyIPSecVpnDpdProfileUpdate,
+		Delete: resourceNsxtPolicyIPSecVpnDpdProfileDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"nsx_id":       getNsxIDSchema(),
+			"path":         getPathSchema(),
+			"display_name": getDisplayNameSchema(),
+			"description":  getDescriptionSchema(),
+			"revision":     getRevisionSchema(),
+			"tag":          getTagsSchema(),
+			"dpd_probe_interval": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  60,
+			},
+			"dpd_probe_mode": {
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice(iPSecVpnDpdProfileDpdProbeModeValues, false),
+				Optional:     true,
+				Default:      model.IPSecVpnDpdProfile_DPD_PROBE_MODE_PERIODIC,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"retry_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  10,
+			},
+		},
+	}
+}
+
+func resourceNsxtPolicyIPSecVpnDpdProfileExists(id string, connector *client.RestConnector, isGlobalManager bool) (bool, error) {
+	client := infra.NewIpsecVpnDpdProfilesClient(connector)
+	_, err := client.Get(id)
+	if err == nil {
+		return true, nil
+	}
+
+	if isNotFoundError(err) {
+		return false, nil
+	}
+
+	return false, logAPIError("Error retrieving resource", err)
```

Comment:
shall we specify resource type?

Reply:
terraform would inform what resource is causing the error, so I think we can remove resource type from other places as well..